tools/wptserve/wptserve/response.py

```diff
@@ -465,9 +465,11 @@ def encode(self, data):
             raise ValueError
 
     def flush(self):
-        """Flush the output."""
+        """Flush the output. Returns False if the flush failed due to
+        the socket being closed by the remote end."""
         try:
             self._wfile.flush()
+            return True
         except socket.error:
             # This can happen if the socket got closed by the remote end
-            pass
+            return False
```

Comment:
@jgraham I want to use this in other tests too, eg to stop generating content when the client is no longer interested.